src/lib/core/components/filter/HistogramFilter.tsx

```diff
@@ -140,7 +145,14 @@ export function HistogramFilter(props: Props) {
     [dataClient, entity, filters, studyId, variable]
   );
   const data = usePromise(
-    useCallback(() => getData(uiState), [getData, uiState])
+    useCallback(() => getData(uiState), [
+      getData,
+      uiState.binWidth,
+      uiState.binWidthTimeUnit,
+      uiState.independentAxisRange,
+    ])
+    // is there some more concise utility to remove a key or keys from an object?
```

Comment:
is there some more concise utility to remove a key or keys from an object?
I tried lodash.omit and it created an endless loop of API calls...

Another option would be to add some structure to uiState: split it into back-end and front-end pieces.